gateleen-cache/src/main/java/org/swisspush/gateleen/cache/fetch/DefaultCacheDataFetcher.java

```diff
@@ -44,48 +45,59 @@ public DefaultCacheDataFetcher(ClientRequestCreator clientRequestCreator, String
     }
 
     @Override
-    public Future<Result<Buffer, StatusCode>> fetchData(final String requestUri, MultiMap requestHeaders, long requestTimeoutMs) {
-        Future<Result<Buffer, StatusCode>> future = Future.future();
+    public Future<Result<Buffer, StatusCode>> fetchData(final String requestUri, HeadersMultiMap requestHeaders, long requestTimeoutMs) {
+        Promise<Result<Buffer, StatusCode>> promise = Promise.promise();
 
         requestHeaders.remove(cacheControlHeader);
-
-        HttpClientRequest fetchRequest = clientRequestCreator.createClientRequest(
+        clientRequestCreator.createClientRequest(
                 HttpMethod.GET,
                 requestUri,
                 requestHeaders,
-                requestTimeoutMs,
-                cRes -> cRes.bodyHandler(data -> {
+                requestTimeoutMs, event -> {
+                    log.warn("Got an error while fetching cache data", event);
+                    promise.complete(Result.err(StatusCode.INTERNAL_SERVER_ERROR));
+                }).onComplete(event -> {
+            if (event.failed()) {
```

Comment:
Why this strange indentation?

Reply:
The default auto-formatter in Intellij formats like this.